File: pizza_flutter/lib/models/order.dart
class OrderItemModel {
  final String productName;
  final String productImageUrl;
  final int quantity;
  final double unitPrice;
  final String size;

  OrderItemModel({
    required this.productName,
    required this.productImageUrl,
    required this.quantity,
    required this.unitPrice,
    required this.size,
  });

  factory OrderItemModel.fromJson(Map<String, dynamic> json) => OrderItemModel(
    productName: json['productName'] ?? '',
    productImageUrl: json['productImageUrl'] ?? '',
    quantity: json['quantity'] ?? 0,
    unitPrice: (json['unitPrice'] as num?)?.toDouble() ?? 0,
    size: json['size'] ?? 'M',
  );
}

class OrderModel {
  final String id;
  final double totalPrice;
  final String status;
  final String paymentStatus;
  final String paymentUrl;
  final String deliveryAddress;
  final DateTime? createdAt;
  final List<OrderItemModel> items;

  OrderModel({
    required this.id,
    required this.totalPrice,
    required this.status,
    required this.paymentStatus,
    required this.paymentUrl,
    required this.deliveryAddress,
    required this.createdAt,
    required this.items,
  });

  bool get isUnpaid => paymentStatus == 'Unpaid';

  factory OrderModel.fromJson(Map<String, dynamic> json) => OrderModel(
    id: json['id']?.toString() ?? '',
    totalPrice: (json['totalPrice'] as num?)?.toDouble() ?? 0,
    status: json['status'] ?? '',
    paymentStatus: json['paymentStatus'] ?? '',
    paymentUrl: json['paymentUrl'] ?? '',
    deliveryAddress: json['deliveryAddress'] ?? '',
    createdAt: DateTime.tryParse(json['createdAt'] ?? ''),
    items: (json['items'] as List? ?? [])
        .map((e) => OrderItemModel.fromJson(e))
        .toList(),
  );
}

File: backend/src/Services/Order/PizzaApp.Order.Core/Entities/Order.cs
namespace PizzaApp.Order.Core.Entities;

public class Order
{
    public string Id { get; set; } = string.Empty;
    public string UserId { get; set; } = string.Empty;
    public List<OrderItem> OrderItems { get; set; } = new();
    public decimal TotalPrice { get; set; }
    public string Status { get; set; } = "AwaitingPayment";
    public string PaymentStatus { get; set; } = "Unpaid";
    public string PaymentMethod { get; set; } = "QR";
    public string PaymentUrl { get; set; } = string.Empty;  // URL xác nhận (mã trong QR)
    public string PaymentQr { get; set; } = string.Empty;   // ảnh QR data URI (denormalized để hiển thị lại)
    public string DeliveryAddress { get; set; } = string.Empty;
    public string ShipperId { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

public class OrderItem
{
    public string ProductId { get; set; } = string.Empty;
    public string ProductName { get; set; } = string.Empty;
    public string ProductImageUrl { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public string Size { get; set; } = "M";
}
